05.wc/wcc.rb

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+WIDTH_SUM_VALUE = 8
+
+def main(args)
+  options = {}
+  options[:filepaths] = args[:filepaths] || args[:filepaths] = false
+  options[:l]         = args[:l] || args[:l] = false
+  options[:w]         = args[:w] || args[:w] = false
+  options[:c]         = args[:c] || args[:c] = false
+
+  if options[:filepaths].empty?
+    show_totals_pipe_input(options)
+  else
+    show_totals_file_input(options)
+  end
+end
+
+def show_totals_pipe_input(options)
+  input_totals = count_input(readlines.join)
+  show_result(input_totals, options)
+end
+
+def show_totals_file_input(options)
+  input_totals_list = []
+  options[:filepaths].each do |fp|
+    input_totals = count_input(File.open(fp).readlines.join)
```

Comment:
```suggestion
    input_totals = count_input(File.read(fp))
```

で済みそうです。

あと、totalという名前を付けると、合計したあとの値が入ってそうな印象を受けます。
左辺と右辺の名前も違うので、そのあたりからもっと良い命名を考えてみてください。

[ユーザーを見つけたら……Aさん「自転車に入れます」Bさん「人間に入れます」Cさん「ユーザーに入れます」 \| FBC](https://bootcamp.fjord.jp/pages/readable-variable-name)

Reply:
readメソッドを教えていただきありがとうございます。修正しました。

変数名は input_stats（← input_totals）で修正しました。